frontend/aml_dashboard/src/index.js

```diff
@@ -0,0 +1,746 @@
+    throwError(new Error('No AML model has been trained'));
+    setTimeout(() => {
+      togAML.$checked.set(false);
+    }, 500);
+  }
+});
+
+const predictionStreamNN = hand_window
+  .filter(() => togNN.$checked.get() && classifier.ready)
+  .map(async (arr) => classifier.predict(calculate_time_features(arr).x))
+  .awaitPromises();
+
+const predictionStreamAML = hand_window
+  .filter(() => togAML.$checked.get() && aml_model['ready'])
+  .map(async (arr) => {
+
+    // Prepare the data for the AML model
+    const json_dt = {data : calculate_time_features(arr).x};
+
+    // Predict using the AML model
+    console.log('Predicting using AML model');
+
+    return fetch(url_inference, {
+      method: "POST", // *GET, POST, PUT, DELETE, etc.
+      mode: "cors", // no-cors, *cors, same-origin
+      cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
+      credentials: "omit", // include, *same-origin, omit
+      headers: {
+        "Content-Type": "application/json",
+      },
+      redirect: "follow", // manual, *follow, error
+      referrerPolicy: "no-referrer", // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
+      body: JSON.stringify(json_dt), // body data type must match "Content-Type" header
+    })
+    .then(response => response.json())
+    .then(async json => {
+      console.log(json);
+      console.log('INFERENCE RECEIVED');
+
+      return json;
+    });
+  })
+  .awaitPromises();
+
+const plotResultsNN = confidencePlot(predictionStreamNN);
+plotResultsNN.title = 'Results Neural Network';
+
+const plotResultsAML = confidencePlot(predictionStreamAML);
+plotResultsAML.title = 'Results AML';
+
+
+// -----------------------------------------------------------
+// INSTANCE VIEWER
+// -----------------------------------------------------------
+
+const create_fiware = button('Create');
+create_fiware.title = 'Fiware Node';
+
+// Configure the parameters for the Fiware Node
+const fiware_params = {
+  parameters: {
+    'Name': new Stream('MyFiwareNode', true),
+    'Domain': new Stream(0, true),
+    'Server IP': new Stream('192.168.1.51', true),
+    'Server Port': new Stream(1028, true),
+    'Context Broker IP': new Stream('localhost', true),
+    'Context Broker Port': new Stream(1026, true),
+    'Entity ID': new Stream('urn:ngsi-ld:Device:001', true),
+    'Attribute Data': new Stream('Data', true),
+    'Attribute Solution': new Stream('Inference', true),
+  },
+};
+
+const fiware_node = modelParameters(fiware_params);
+fiware_node.title = 'Choose parameters for the Fiware Node creation:';
+
+const fiware_node_status = text('Not Created');
+fiware_node_status.title = 'Fiware Node Status';
+
+const post_data = button('Post Data');
+post_data.title = 'Context Broker Data';
+
+const upload_data = imageUpload();
+upload_data.title = 'Upload data to the Context Broker';
```

Comment:
I would recommend changing the order of the sections to make the use more intuitive, following the logical order of first uploading the data to the context broker and then posting it.
```suggestion

const upload_data = imageUpload();
upload_data.title = 'Upload data to the Context Broker';

const post_data = button('Post Data');
post_data.title = 'Context Broker Data';


```

Reply:
When changing order,  abnormal behaviour airse, such as displaying a result when there has been no inference done yet.